src/main/kotlin/BUILD

```diff
@@ -48,13 +49,50 @@ java_binary(
         ":compiler_lib.jar",
         "@com_github_jetbrains_kotlin//:lib/kotlin-compiler.jar",
     ],
-    main_class = "io.bazel.kotlin.builder.KotlinBuilderMain",
-    runtime_deps = [":builder_jar_jar"],
     jvm_flags = [
         "-XX:+IgnoreUnrecognizedVMOptions",
         "--add-opens=java.base/java.nio=ALL-UNNAMED",
         "--add-opens=java.base/java.lang=ALL-UNNAMED",
         "--add-opens=jdk.jdeps/com.sun.tools.jdeps=ALL-UNNAMED",
     ],
+    main_class = "io.bazel.kotlin.builder.KotlinBuilderMain",
     visibility = ["//visibility:public"],
-)
\ No newline at end of file
+    runtime_deps = [":builder_jar_jar"],
+)
+
+release_archive(
+    name = "pkg",
+    srcs = [
+        ":compiler_lib.jar",
+    ],
+    # Generate the release build file, relying on precompiled jars.
+    dist_files = {
+        "BUILD": """
```

Comment:
Nit: You might want to break these build files into private vars, like _RELEASE_LIB_BUILD_CONTENT or something.  I find it makes the whole thing read more cleanly if it's done that way, because you don't end up with weird scoping/indenting oddities.

Reply:
Made them separate files.